src/common/lib/lnurl.ts

```diff
@@ -61,27 +68,51 @@ const lnurl = {
 
     return null;
   },
+
   async getDetails(lnurlString: string): Promise<LNURLError | LNURLDetails> {
     const url = normalizeLnurl(lnurlString);
-    let lnurlDetails = {} as LNURLDetails;
-    lnurlDetails.tag = url.searchParams.get("tag") as LNURLDetails["tag"];
-    if (lnurlDetails.tag === "login") {
-      lnurlDetails.k1 = url.searchParams.get("k1") || "";
-      lnurlDetails.action = url.searchParams.get("action") || undefined;
+    const searchParamstag = url.searchParams.get("tag") as LNURLDetails["tag"];
+
+    if (searchParamstag === "login") {
+      const searchParamsK1 = url.searchParams.get(
+        "k1"
+      ) as LNURLAuthServiceResponse["tag"];
+      const searchParamsAction = url.searchParams.get(
+        "action"
+      ) as LNURLAuthServiceResponse["tag"];
+
+      const lnurlAuthDetails: LNURLAuthServiceResponse = {
+        action: searchParamsAction,
+        domain: url.hostname,
+        k1: searchParamsK1,
+        tag: searchParamstag,
+        url: url.toString(),
+      };
+
+      return lnurlAuthDetails;
```

Comment:
Same here: Rather remove the assertion as the searchparam ca be `null`
Which is actually the case when copying the LNURL Auth from https://lnurl.fiatjaf.com/
=> `action` is `null`

But `k1` has to be defined, right? Otherwise the URL is broken, correct?
Let´s test this and throw otherwise.

here is my suggestion:

```TS
    if (searchParamstag === "login") {
      const searchParamsK1 = url.searchParams.get("k1");
      const searchParamsAction = url.searchParams.get("action");

      if (searchParamsK1) {
        const lnurlAuthDetails: LNURLAuthServiceResponse = {
          action: searchParamsAction,
          domain: url.hostname,
          k1: searchParamsK1,
          tag: searchParamstag,
          url: url.toString(),
        };

        return lnurlAuthDetails;
      }

      throw new Error(
        "LNURL AUTH Login is missing search params: searchParamsK1"
      );
     } else {
     ...
```

Reply:
This would then require to change these types too:

https://github.com/getAlby/lightning-browser-extension/blob/master/src/types.ts#L347
=> `action: string | null:`

https://github.com/getAlby/lightning-browser-extension/blob/master/src/types.ts#L293-L296
=> add `action: string | null`

Reply:
Yes, makes sense. `action` is defined as optional in the spec:  
> optional action enum which can be one of four strings: register | login | link | auth.

https://github.com/fiatjaf/lnurl-rfc/blob/luds/04.md#wallet-to-service-interaction-flow

Reply:
Updated it with your suggestion. Do we need to add ` | null` to the types if they are already optional (`action?: ...`)?

Reply:
Either decide for:

- action is _not_ optional and can be a string or null
- or action is optional and can be a string (or undefined)

The first case happens when you just pass on the data from the LNURL url.

The latter is when you decide what to do (your current code [here](https://github.com/getAlby/lightning-browser-extension/pull/1545/files#diff-146ca15116557d76017757b844192427543e8137310612ff1d0b5e7221270187R82)).

As the lnurlDetails for LNURLAuth are always constructed from SearchParams of an URL (correct?), 
I would go for the latter case
`action: string | null;` and paste the link to the fjatfjat docs next to the types.

Reply:
Got it! Corrected it.